File: spring-graphql/src/main/java/com/example/demo/gql/directives/UpperCaseDirectiveWiring.java
package com.example.demo.gql.directives;

import graphql.schema.DataFetcherFactories;
import graphql.schema.GraphQLFieldDefinition;
import graphql.schema.idl.SchemaDirectiveWiring;
import graphql.schema.idl.SchemaDirectiveWiringEnvironment;

public class UpperCaseDirectiveWiring implements SchemaDirectiveWiring {
    @Override
    public GraphQLFieldDefinition onField(SchemaDirectiveWiringEnvironment<GraphQLFieldDefinition> env) {

        var field = env.getElement();
        //var parentType = env.getFieldsContainer();
        //var originalDataFetcher = env.getCodeRegistry().getDataFetcher(parentType, field);
        var dataFetcher = DataFetcherFactories.wrapDataFetcher(env.getFieldDataFetcher(),
                (dataFetchingEnvironment, value) -> {
                    if (value instanceof String s) {
                        return s.toUpperCase();
                    }
                    return value;
                }
        );

        //env.getCodeRegistry().dataFetcher(FieldCoordinates.coordinates(parentType, field), dataFetcher);
        env.setFieldDataFetcher(dataFetcher);
        return field;
    }
}

File: graphql-kotlin/src/main/kotlin/com/example/demo/gql/directives/UpperCaseDirectiveWiring.kt
package com.example.demo.gql.directives

import com.expediagroup.graphql.generator.directives.KotlinFieldDirectiveEnvironment
import com.expediagroup.graphql.generator.directives.KotlinSchemaDirectiveWiring
import graphql.schema.DataFetcherFactories
import graphql.schema.GraphQLFieldDefinition
import java.util.*

class UpperCaseDirectiveWiring : KotlinSchemaDirectiveWiring {
    override fun onField(env: KotlinFieldDirectiveEnvironment): GraphQLFieldDefinition {

        val field = env.element
        val dataFetcher = DataFetcherFactories
            .wrapDataFetcher(
                env.getDataFetcher()
            ) { _, value -> if (value is String && value.isNotEmpty()) value.uppercase(Locale.getDefault()) else value }

        env.setDataFetcher(dataFetcher);
        return field
    }
}
